src/transformers/models/llama/modeling_llama.py

```diff
@@ -81,14 +81,12 @@ def __init__(self, hidden_size, eps=1e-6):
         self.variance_epsilon = eps
 
     def forward(self, hidden_states):
+        input_dtype = hidden_states.dtype
         variance = hidden_states.to(torch.float32).pow(2).mean(-1, keepdim=True)
         hidden_states = hidden_states * torch.rsqrt(variance + self.variance_epsilon)
 
-        # convert into half-precision if necessary
-        if self.weight.dtype in [torch.float16, torch.bfloat16]:
-            hidden_states = hidden_states.to(self.weight.dtype)
-
-        return self.weight * hidden_states
+        hidden_states = (self.weight * hidden_states).to(input_dtype)
+        return hidden_states
```

Comment:
The Llama modeling file should not be touched by this PR. If this change is necessary, it should go in its own PR.